packages/connect/e2e/tests/device/cancelCoinjoinAuthorization.test.ts

```diff
@@ -0,0 +1,98 @@
+/* eslint-disable import/no-named-as-default */
+
+import TrezorConnect, { Success, PROTO, Unsuccessful } from '../../../src';
+
+const { ADDRESS_N } = global.TestUtils;
+const { getController, setup, conditionalTest, initTrezorConnect } = global.Trezor;
+
+const controller = getController('applyFlags');
+
+describe('TrezorConnect.cancelCoinjoinAuthorization', () => {
+    beforeAll(async () => {
+        await setup(controller, {
+            mnemonic: 'mnemonic_all',
+        });
+    });
+
+    beforeEach(async () => {
+        // restart connect for each test (working with event listeners)
+        await TrezorConnect.dispose();
+
+        await initTrezorConnect(controller, { debug: false });
+
+        TrezorConnect.on('DEVICE_EVENT', ev => {
+            if ('code' in ev.payload && ev.payload.code === 'ButtonRequest_PinEntry') {
+                controller.send({ type: 'emulator-input', value: '1234' });
+            }
+        });
+    });
+
+    afterAll(async () => {
+        controller.dispose();
+        await TrezorConnect.dispose();
+    });
+
+    conditionalTest(['1', '<2.5.4'], 'Cancel authorization works', async () => {
+        // const result = await TrezorConnect.cancelCoinjoinAuthorization({});
+
+        // expect(result.success).toBe(true);
+        // expect((result as Success<PROTO.Success>).payload.message).toBe('Authorization cancelled');
+
+        //
+        //
+        //
+        //
+
+        // unlocked path is required for tx validation
+        const unlockPath = await TrezorConnect.unlockPath({
+            path: ADDRESS_N("m/10025'"),
+        });
+
+        if (!unlockPath.success) throw new Error(unlockPath.payload.error);
+
+        const auth = await TrezorConnect.authorizeCoinjoin({
+            coordinator: 'www.example.com',
+            maxRounds: 2,
+            maxCoordinatorFeeRate: 500000, // 5% => 0.005 * 10**8;
+            maxFeePerKvbyte: 3500,
+            path: ADDRESS_N("m/10025'/1'/0'/1'"),
+            coin: 'Testnet',
+            scriptType: 'SPENDTAPROOT',
+        });
+
+        expect(auth.success).toBe(true);
+        expect(auth.payload).toEqual({ message: 'Coinjoin authorized' });
+
+        const commitmentData =
+            '0f7777772e6578616d706c652e636f6d0000000000000000000000000000000000000000000000000000000000000001';
+
+        const proof = await TrezorConnect.getOwnershipProof({
+            coin: 'Testnet',
+            path: ADDRESS_N("m/10025'/1'/0'/1'/1/0"),
+            scriptType: 'SPENDTAPROOT',
+            userConfirmation: true, // ButtonRequest is not emitted because of preauthorization
```

Comment:
pls remove comment from here and line 90, it's misleading since you are not testing button requests

Reply:
5780f7c05